R/summarise_by_reference_time.R

```diff
@@ -0,0 +1,43 @@
+#' Summarise a dataframe containing draws of reporting squares
+#'
+#' This function ingests a long tidy dataframe containing many draws of
+#'    reporting squares across each reference time, by summing over all the
+#'    counts reported at each delay `d`.
+#'
+#' @param nowcast_draws_df Dataframe containing draws of a reporting square.
+#'    Must contain the following columns: `time`, `delay`, `draw`, `count`.
+#'
+#' @returns `summary_df` Dataframe containing the nowcasted expected observed
+#'    total observations at each reference time.
+#' @export
+#' @importFrom checkmate check_names
+#'
+#' @examples
+#' nowcast_draws_df <- data.frame(
+#'   time = c(1, 1, 1, 1, 2, 2, 2, 2),
+#'   delay = c(1, 2, 1, 2, 1, 2, 1, 2),
+#'   draw = c(1, 1, 2, 2, 1, 1, 2, 2),
+#'   count = c(3, 6, 4, 7, 1, 2, 2, 3)
+#' )
+#' summary_df <- summarise_by_reference_time(nowcast_draws_df)
+#' print(summary_df)
+summarise_by_reference_time <- function(nowcast_draws_df) {
```

Comment:
I think this is a bit opaque in terms of what it actually does which is sum counts to what is reported/nowcast. I think this comes from sticking a bit closely to the summarise approach.

Traditionally what this function is putting out is the nowcast so that could be something to ground on. 

Just checking but the input here has already been fused so it is the observed data where it is available and the nowcast where it is not?

Reply:
Yes it is the observed and nowcated data fused together. 

I am not entirely sure we need this function, as it is basically just another implementation of `dply::summarise`?

Or is your suggestion to just call this something more like `get_nowcast_df()`...

Reply:
changed name to `get_nowcast_df()`